src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,274 @@
+        }
+      }
+    }
+    return strwords[0];
+    file.close();
+  }
+
+//Array 2 - One Dimensional
+  public int[] notAlone(int[] nums, int value){
+    int intcounter;
+
+    intcounter = 1;
+
+    while(intcounter< nums.length){
+      if (nums[intcounter] == value){
+        if(nums[intcounter-1]!= value){
+          if(nums[intcounter] !=value){
+            if(nums[intcounter -1]>nums[intcounter]){
+              nums[intcounter]= nums[intcounter -1];
+            }
+            if (nums[intcounter + 1]>nums[intcounter]){
+              nums[intcounter] = nums[intcounter+1];
+            }
+          }
+        }
+      }
+      intcounter++;
+    }
+    return nums;
+  }
+
+//Array 4 - One Dimensional - Two Loops
+  public boolean canBalance(int[]num){
+    boolean split;
+    int intcounter;
+    int intcountersmall;
+    int intcounterlarge;
+    int intsumsmall;
+    int intsumlarge;
+
+    intcounter = 1;
+    intsumsmall = 0;
+    intsumlarge = 0;
+    split =false;
+
+    while(intcounter<nums.length && split == false){
+      for(intcountersmall = intcounter; intcountersmall>=0; intcountersmall--){
+        intsumsmall= intsumsmall + num[intcountersmall];
+      }
+      for(intcounterlarge = intcounter + 1; intcounterlarge<=num.length; intcounterlarge++){
+        intsumlarge = intsumlarge + num[intcounterlarge];
+      }
+      if (intsumlarge == intsumsmall){
+        split = true;
+      }
+      intcounter++;
+    }
+    return split;
+  }
+
+  //Array 6 - Two Dimensional
+    public static void diagonal(int n){
+    int square[][];
+    int intcount1;
+    int intcount2;
+    int inthori;
+    int inthori2;
+
+    square = new int[n][n];
+    intcount1 = n;
+    intcount2 = 0;
+    inthori = 0;
+    inthori2=n;
+
+    for(intcount1=n; intcount1>0;intcount1--){
+      square[intcount1][intcount2] = 1;
+      for(inthori=0;inthori<intcount1;inthori++){
+        square[intcount1][inthori]=0;
+      }
+      for(inthori2=n;inthori2>intcount1;inthori2--){
+        square[intcount1][inthori2]=2;
+      }
+    }
+    System.out.println(square);
+  }
+
+
+    /**
+    * remove the middle letter
+    * @param String 1  The first letter
+    * @param String 2  The last letter
+    * @param String 2  Both letters 
+    * @author Adsana 
+    */
+
+
+
+    // Create a method to print out first and last letters of a 3 letter word
+    // Use substring to print out the letters we want
+    public static String zipZap(String strWord){
+      String strLetter1;
+      String strLetter2;
+      String strTwoLetters;
+     
+      strWord = "";
+      strLetter1 = strWord.substring(0);
+      strLetter2 = strWord.substring(2);
+      strTwoLetters = strLetter1 + strLetter2; 
+
+    
+    return strTwoLetters;
+    }
+
+     /**
+    * print longest word in file
+    * @param String 1  The words 
+    * @param String 2  The longest word
+    * @author Adsana 
+    */
     
+    TextInputFile words = new TextInputFile("words.txt");
```

Comment:
@Adsana1221 Fails here. Improper method of how to read a data file, as instructed in class, you need to use
`BufferedReader thefile = new BufferedReader(new FileReader(strFileName));`